lib/src/sketch_schemes/mash.rs

```diff
@@ -66,7 +65,10 @@ impl MashSketcher {
 }
 
 impl SketchScheme for MashSketcher {
-    fn process(&mut self, seq: &SequenceRecord) {
+    fn process<'s, 'a: 's, 'b>(&'a mut self, seq: &'s dyn Sequence<'b>)
```

Comment:
Same lifetime legibility thing if you also find it easier to read, up to you either way though